src/main.js

```diff
@@ -807,18 +809,18 @@ Apify.main(async () => {
                             url: page.url(),
                         });
 
-                        // Extract home data from mapResults
-                        const thr = input.splitThreshold || 500;
+                        log.info(`Found ${results.length} results in current area`, qs.mapBounds);
 
-                        if (results.length >= thr) {
+                        // Extract home data from mapResults
+                        if (zpids.size < totalResults) {
```

Comment:
this is not correct, zillow limit the results to 500-800 items (over 20 pages, most of the time 40 results per page). if there are more than 500 results, it will try to split the map. the current collected zpids has nothing to do with the returned size by their API

Reply:
@pocesar I get the point that map splitting should be used only if there are more than 500 results but when we ran tests with less than 500 results, many items were missed by the scraper even if they seemed to be included in the extracted json (according to the log). I tried to change the behavior to use map splitting even if there are fewer than 500 results (comparing max number of results found with `zpids.size`). This way it forced map splitting whenever we expect more results than we already have.

In my test scenario, expected number of results was 145. It scraped 100 results without using map splitting and 141 results with map splitting used (still not sure where the remaining 4 items got lost). 

The corresponding test input:
```json
{
  "type": "sale",
  "startUrls": [
    {
      "url": "https://www.zillow.com/victoria-bc/?searchQueryState=%7B%22pagination%22%3A%7B%7D%2C%22usersSearchTerm%22%3A%22Victoria%2C%20BC%22%2C%22mapBounds%22%3A%7B%22west%22%3A-123.40896545275879%2C%22east%22%3A-123.30854354724121%2C%22south%22%3A48.40384834627835%2C%22north%22%3A48.44952652618214%7D%2C%22mapZoom%22%3A5%2C%22regionSelection%22%3A%5B%7B%22regionId%22%3A791371%2C%22regionType%22%3A6%7D%5D%2C%22isMapVisible%22%3Atrue%2C%22filterState%22%3A%7B%22ah%22%3A%7B%22value%22%3Atrue%7D%7D%2C%22isListVisible%22%3Atrue%7D",
      "method": "GET"
    }
  ],
  "simple": true,
  "proxyConfiguration": {
    "useApifyProxy": true
  },
  "maxRetries": 20,
  "extendOutputFunction": "async ({ data, item, customData, Apify }) => {\n    return item;\n}",
  "extendScraperFunction": "async ({ label, page, request, customData, Apify }) => {\n    if (label === 'SETUP') {\n        // before crawler.run()\n    } else if (label === 'GOTO') {\n        // inside handleGotoFunction\n    } else if (label === 'HANDLE') {\n        // inside handlePageFunction\n    } else if (label === 'FINISH') {\n        // after crawler.run()\n    }\n}",
  "handlePageTimeoutSecs": 3600,
  "stealth": false,
  "debugLog": false,
  "customData": {}
}
```